### PR TITLE
add support for container volumes

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

File: src/pytest_container/container.py [hunk 1]
```diff
@@ -188,7 +194,8 @@ class ContainerVolume:
     #: Path inside the container where this volume will be mounted
     container_path: str
 
-    #: Path on the host that will be mounted. When omitted, a temporary
+    #: Path on the host that will be mounted if absolute. if relative,
+    #  it refers to a volume to be auto-created. When omitted, a temporary
```
Comment (dcermak) [inline src/pytest_container/container.py:198]:
```suggestion
    #: it refers to a volume to be auto-created. When omitted, a temporary
```

File: src/pytest_container/container.py [hunk 2]
```diff
@@ -280,6 +287,17 @@ def cleanup(self) -> None:
             )
             assert self.host_path
             self._tmpdir.cleanup()
+        else:
+            # Clean up container volume
+            if not os.path.isabs(self.host_path):
```
Comment (dcermak) [inline src/pytest_container/container.py:292]: mypy complaint: Argument 1 to "isabs" has incompatible type "Optional[str]"; expected "Union[str, bytes, PathLike[str], PathLike[bytes]]"

File: src/pytest_container/container.py [hunk 3]
```diff
@@ -280,6 +287,17 @@ def cleanup(self) -> None:
...
+                check_output(
+                    [
+                        container_runtime,
+                        "volume",
+                        "rm", "-f",
+                        self.host_path,
```
Comment (dcermak) [inline src/pytest_container/container.py:298]: mypy complaint: List item 4 has incompatible type "Optional[str]"; expected "Union[str, bytes, PathLike[str], PathLike[bytes]]"

Comment (dirkmueller) [reply src/pytest_container/container.py:298]: I kinda don't get that locally. but I added an assert now, maybe that helps

Comment (dcermak) [reply src/pytest_container/container.py:298]: you only get this warning when you invoke mypy via `nox -s lint`

File: src/pytest_container/container.py (hunk 1 above)
Comment (dirkmueller) [reply src/pytest_container/container.py:198]: added

File: src/pytest_container/container.py (hunk 3 above)
Comment (dirkmueller) [reply src/pytest_container/container.py]: yeah, not in my setup ?!

**Comment by dirkmueller** [comment]
> Please add an integration test for this.

Tests added

**Review by dcermak — APPROVED** [review]
LGTM!

**Comment by dcermak** [comment]
@dirkmueller Can you please rebase?

**Comment by dirkmueller** [comment]
Will do a rebase later today

**Pull request merged**